airflow/providers/cncf/kubernetes/operators/kubernetes_pod.py

```diff
@@ -328,101 +339,133 @@ def create_labels_for_pod(context) -> dict:
-        :type try_numbers_match: bool
-        :param launcher: PodLauncher
-        :param pod: Pod found with matching labels
-        """
-        if try_numbers_match:
-            log_line = f"found a running pod with labels {labels} and the same try_number."
-        else:
-            log_line = f"found a running pod with labels {labels} but a different try_number."
-
-        # In case of failed pods, should reattach the first time, but only once
-        # as the task will have already failed.
-        if self.reattach_on_restart and not pod.metadata.labels.get("already_checked"):
-            log_line += " Will attach to this pod and monitor instead of starting new one"
-            self.log.info(log_line)
-            self.pod = pod
-            final_state, remote_pod, result = self.monitor_launched_pod(launcher, pod)
+        pod_phase = remote_pod.status.phase if hasattr(remote_pod, 'status') else None
+        if pod_phase != PodStatus.SUCCEEDED:
+            if self.log_events_on_failure:
+                with _suppress(Exception):
+                    for event in self.launcher.read_pod_events(pod).items:
+                        self.log.error("Pod Event: %s - %s", event.reason, event.message)
+            if not self.is_delete_operator_pod:
+                with _suppress(Exception):
+                    self.patch_already_checked(pod)
+            raise AirflowException(f'Pod {pod and pod.metadata.name} returned a failure: {remote_pod}')
+
+    def process_pod_deletion(self, pod):
+        if self.is_delete_operator_pod:
+            self.log.info("deleting pod: %s", pod.metadata.name)
+            self.launcher.delete_pod(pod)
         else:
-            log_line += f"creating pod with labels {labels} and launcher {launcher}"
-            self.log.info(log_line)
-            final_state, remote_pod, result = self.create_new_pod_for_operator(labels, launcher)
-        return final_state, remote_pod, result
+            self.log.info("skipping deleting pod: %s", pod.metadata.name)
 
-    @staticmethod
-    def _get_pod_identifying_label_string(labels) -> str:
+    def _get_pod_identifying_label_string(self, labels) -> str:
         label_strings = [
             f'{label_id}={label}' for label_id, label in sorted(labels.items()) if label_id != 'try_number'
         ]
-        return ','.join(label_strings) + ',already_checked!=True'
-
-    @staticmethod
-    def _try_numbers_match(context, pod) -> bool:
-        return pod.metadata.labels['try_number'] == context['ti'].try_number
+        return ','.join(label_strings) + f',{self.POD_CHECKED_KEY}!=True'
+
+    def _compare_try_numbers(self, context, pod):
+        tries_match = pod.metadata.labels['try_number'] == context['ti'].try_number
+        self.log.info(
+            "found a running pod with labels %s %s try_number.",
+            pod.metadata.labels,
+            "and the same" if tries_match else "but a different",
+        )
```

Comment:
This method name and the argument names read unnaturally to me. How about only passing in the try numbers instead of the complete objects, so that it's clear from just the method name what's expected:

```suggestion
    def _compare_try_numbers(self, context_try_number: int, pod_try_number: int):
        tries_match = context_try_number == pod_try_number
        self.log.info(
            "Found a running pod with labels %s %s try_number.",
            pod.metadata.labels,
            "and the same" if tries_match else "but a different",
        )
```

And calling it as:
```python
self._compare_try_numbers(context['ti'].try_number, pod.metadata.labels['try_number'])
```

Reply:
i think i can actually chop this method and inline the code.

i think maybe at an earlier stage, there may have been too much going on at the call site and i wanted to get more logic out but looking again now it's more manageable.

since we're talking about it, i'm not reallly sure why we _care_ whether the tries match.  we don't _do_ anything with that information apart from logging it.  it seems more appropriate as debug info?   @dimberman or @jedcunningham, you think it's ok to just chop this logic altogether?

Reply:
update: i inlined the comparison and simplified it a bit just plainly stating the facts instead of the flowery expression

```python
            self.log.info("Found matching pod %s with labels %s", pod.metadata.name, pod.metadata.labels)
            self.log.info("`try_number` of task_instance: %s", context['ti'].try_number)
            self.log.info("`try_number` of pod: %s", pod.metadata.labels['try_number'])
```